python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+                    err = ("`array` and each item in `singles` must have the same spatial dimension, "
+                           "received dimensions {0} for array with dimension {1}")
+                    raise ValueError(err.format([single.array.shape for single in singles], array.shape))
+                self._singles = tuple(singles)
+                self._bbox = singles[0].getBBox()
+        else:
+            # Extract the single band objects and filters
+            if singles is None and filters is not None:
+                # Attempt to load a set of images
+                self._singleType = singleType
+                singles = []
+                for f in filters:
+                    if filterKwargs is not None:
+                        for key, value in filterKwargs:
+                            kwargs[key] = value[f]
+                    singles.append(singleType(**kwargs))
+            if singles is not None:
+                super().__init__(filters, singles)
+            else:
+                err = ("Either a list of `singles` and `filters`; "
+                       "or an `array`, `filters`, `bbox`, and `singleType`; "
+                       "or a set of `kwargs` is required to initialize a `MultibandImage`.")
+                raise NotImplementedError(err)
+
+            self._array = np.array([image.array for image in self.singles])
+            self._updateSingles(type(self.singles[0]))
+
+        # Make sure that all of the parameters have been setup appropriately
+        if not isinstance(self._bbox, Box2I):
+            raise RuntimeError("Something went wrong, `self._bbox` should be a `Box2I`")
+        if not np.all([single.getBBox() == self.getBBox() for single in self.singles]):
+            raise ValueError("Single band images did not all have the same bounding box")
+
+    def _getArray(self):
+        """Data cube array in multiple bands
+        """
+        return self._array
+
+    def _setArray(self, value):
+        self.array[:] = value
+
+    array = property(_getArray, _setArray)
+
+    def _updateSingles(self, singleType):
+        """Update the Image<X> in each band
+
+        This method is called when a `MultibandImage` is initialized.
+        """
+        self._singleType = singleType
+        self._singles = self._arrayToSingles(self.array, singleType, self.getXY0())
+        if not len(self.filters) == len(self.singles):
+            raise RuntimeError("The length of `filters` and `array` should be the same")
+
+    def _arrayToSingles(self, array, singleType, xy0):
+        """Create a list of `Image<X>` objects from a 3D array
+        """
+        return tuple([singleType(array=array[n], xy0=xy0) for n in range(len(array))])
+
+    def copy(self, deep=False):
+        """Copy the current object
+
+        Parameters
+        ----------
+        deep: bool
+            Whether or not to make a deep copy
```

Comment:
These are all good points. I didn't notice that `clone` was added because it isn't wrapped in the typical pybind11 files (eg image.cc), which I noticed defaults to `deep=True`. So I renamed `copy` to `clone` and made the default `deep=True` for consistency.